### PR TITLE
Fixes #20026

### LINKED ISSUE #20026: Imperfect error message for invoking null literal as function
**TypeScript Version:** 2.6.1

**Code**

Minimal:
```ts
null()
```

Actual real-world mistake:

```tsx
str !== 'hello' && thing !== null (
  <div>blah</div>
)
```

**Expected behavior:**

Clear error to the effect of:

> Invoking 'null' as a function

**Actual behavior:**

Technically correct but slightly misleading error:

> Object is possibly 'null'

### PR DESCRIPTION
**Pull request by remojansen**
Here's a checklist you might find useful.
- [x] There is an associated issue that is labelled
  'Bug' or 'help wanted' or is in the Community milestone
- [x] Code is up-to-date with the `master` branch
- [x] You've successfully run `jake runtests` locally
- [x] You've signed the CLA
- [x] There are new or updated unit tests validating the change

Fixes #20026

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -15539,6 +15539,20 @@ namespace ts {
             return checkNonNullType(checkExpression(node), node);
         }
 
+        function checkNonNullCallLikeExpression(node: Expression | QualifiedName) {
```
Comment (mhegazy) [inline src/compiler/checker.ts]: i do not think we should duplicate the function implementation here. consider passing the new error messages as optional arguments to checkNullExpression instead.

Comment (remojansen) [reply src/compiler/checker.ts]: @mhegazy thanks for the code review. I just sent a new commit that uses optional arguments instead of a new function.

**Comment by remojansen** [comment]
Hi @mhegazy I just resolved some merge conflicts. Is there anything else that I can do to speed up merging this PR into master?

**Comment by mhegazy** [comment]
Sorry for the delay, can you please update your PR.

**Comment by remojansen** [comment]
@mhegazy thanks for merging the PR, can you please explain to me what was that timeout error in the build? I would like to learn about it future contributions. Thanks!

**Comment by mhegazy** [comment]
`@types/mocha` changed, and we needed to respond to that. see https://github.com/Microsoft/TypeScript/pull/21075 for more details.

**Pull request merged**